packages/smooth_app/lib/pages/onboarding/country_selector.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/material.dart';
+import 'package:iso_countries/iso_countries.dart';
+import 'package:openfoodfacts/utils/CountryHelper.dart';
+import 'package:smooth_app/data_models/user_preferences.dart';
+import 'package:smooth_ui_library/util/ui_helpers.dart';
+
+/// A selector for selecting user's country.
+class CountrySelector extends StatefulWidget {
+  const CountrySelector();
+
+  @override
+  State<CountrySelector> createState() => _CountrySelectorState();
+}
+
+class _CountrySelectorState extends State<CountrySelector> {
+  late UserPreferences _userPreferences;
+  late List<Country> _countryList = <Country>[];
+  late String _chosenValue;
+  late Future<void> _initFuture;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _initFuture = _init();
```

Comment:
I would call it rather in `initState`, but perhaps you have a good reason to call it here.

Reply:
I get this error in initState:

`Unhandled Exception: inheritFromWidgetOfExactType(_LocalizationsScope) or inheritFromElement() was called before initState() completed`